src/include/migraphx/op/resize.hpp

```diff
@@ -0,0 +1,331 @@
+            // No size/scale input.  Size/scale must be an attribute and so output will be static.
+            if((sizes.empty()) == (scales.empty()))
+                MIGRAPHX_THROW(
+                    "RESIZE: One and only one of sizes or scales attributes must be given");
+            if(not sizes.empty())
+            {
+                if(not(sizes.size() == input_s.ndim()))
+                    MIGRAPHX_THROW("RESIZE: sizes attribute's size must match rank of input X");
+                std::vector<size_t> lens;
+                std::transform(sizes.begin(),
+                               sizes.end(),
+                               std::back_inserter(lens),
+                               [](auto in_len) { return static_cast<size_t>(in_len); });
+                return shape{input_s.type(), lens};
+            }
+            else
+            {
+                if(not(scales.size() == input_s.ndim()))
+                    MIGRAPHX_THROW("RESIZE: scales attribute's size must match rank of input X");
+                std::vector<size_t> lens;
+                std::transform(scales.begin(),
+                               scales.end(),
+                               input_s.lens().begin(),
+                               std::back_inserter(lens),
+                               [](auto scale_i, size_t in_len) {
+                                   return static_cast<size_t>(scale_i * in_len);
+                               });
+                return shape{input_s.type(), lens};
+            }
+        }
+        else
+        {
+            // Dynamic output shape.
+            if(inputs.back().ndim() != 1)
+                MIGRAPHX_THROW("RESIZE: size/scale input must have rank 1");
+            if(inputs.back().dynamic() and not inputs.back().dyn_dims()[0].is_fixed())
+                MIGRAPHX_THROW("RESIZE: size/scale input must be fixed size");
+
+            if(inputs.front().ndim() != inputs.back().to_static(1).lens()[0])
+                MIGRAPHX_THROW("RESIZE: size/scale input's size must match rank of input X");
+
+            // The output shape is dynamic, with an unlimited size range.
+            std::size_t max_val = std::numeric_limits<std::size_t>::max();
+            std::vector<shape::dynamic_dimension> dyn_dims(inputs.back().lens().at(0),
+                                                           shape::dynamic_dimension{0, max_val});
+            return {inputs.front().type(), dyn_dims};
+        }
+    }
+
+    argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
+    {
+        shape output_shape;
+        auto in_lens = args[0].get_shape().to_static(1).lens();
+        std::vector<size_t> out_lens(in_lens.size());
+
+        // Scales are either given, or calculated from output shape
+        std::vector<float> vec_scale(in_lens.size(), 1.0f);
+
+        if(dyn_out.computed_shape.dynamic() and args.size() > 1)
+        {
+            args[1].visit([&](auto input) {
```

Comment:
Need to look at the values inside of `scales` or `sizes` so cannot use the `dyn_output` struct.

Reply:
I'm not sure any significant improvement is possible.  Either shapes or scales can be specified, and each can appear either as an attribute (constant) or as an input (runtime).  So the handling logic has to have lots of permutations.  If you can think of something that's clearer, I'd like to hear it but any change may be error-prone.

Reply:
* Do not use `dyn_output.computed_shape` it reruns compute_shape() with the shapes of the input arguments.
* Have two cases one with 1 input argument (constant scales or sizes) and one with 2 input arguments (variable scales or sizes).
* You can directly use the output shape calculated by compute_shape() for the 1 input argument version since this is the static shape case. 
* All dynamic shape cases that are used in current models will have >=2 inputs. In this case you have to calculate the output shape at runtime with the values in scales or sizes.

Reply:
ok, this change is made although the difference seems insignificant.